yjit/bindgen/src/main.rs

```diff
@@ -0,0 +1,65 @@
+//! This is the binding generation tool that the YJIT cruby module talks about.
+//! More docs later once we have more experience with this, for now, check
+//! the output to make sure it looks reasonable and allowlist things you want
+//! to use in Rust.
+extern crate bindgen;
+
+use std::path::PathBuf;
+use std::env;
+
+fn main() {
+    // Remove this flag so rust-bindgen generates bindings
+    // that are internal functions not public in libruby
+    let filtered_clang_args = env::args().filter(|arg| arg != "-fvisibility=hidden");
+
+    // assume CWD is Ruby repo root so we could copy paste include path
+    // args from make.
+    let bindings = bindgen::builder()
+        .clang_args(filtered_clang_args)
+        .header("internal.h")
+
+        // Some C functions that were expressly for Rust YJIT in this
+        // file. TODO: Might want to move them later.
+        .header("yjit.c")
+
+        // Don't want to copy over C comment
+        .generate_comments(false)
+
+        // Don't want layout tests as they are platform dependent
+        .layout_tests(false)
+
+        // This struct is public to Ruby C extensions
+        .allowlist_type("RBasic")
+
+        .allowlist_function("rb_hash_new")
+        .allowlist_function("rb_hash_aset")
+
+        .allowlist_function("rb_iseq_(get|set)_yjit_payload")
+
+        .allowlist_function("rb_iseq_pc_at_idx")
+        .allowlist_function("rb_iseq_opcode_at_pc")
```

Comment:
This looks like we add C bindings individually here. Are we planning to do more of a mass import, or are we expecting to basically have a "symbols we use from CRuby" list here, longer-term?

Reply:
It's probably going to be a mix. Since regex is supported here we could bind things that are expressly for YJIT in one go by following some naming convention, but we'll bind some other functions we don't write.